spec/requests/courses_spec.rb

```diff
@@ -18,24 +18,24 @@
   # Course. As you add validations to Course, be sure to
   # adjust the attributes here as well.
 
-  let!(:user) { User.create(email: "user@test.com", password: "testpass") }
-  let(:subject) { Subject.create!(name: "subject name", user: user) }
-  let(:valid_attributes) { { name: "course name", subject: subject } }
-  let(:invalid_attributes) { { name: nil, subject: subject } }
+  let!(:user) { FactoryBot.create(:user) }
+  let!(:subject) { FactoryBot.create(:subject, user: user) }
+  let!(:valid_attributes) { { name: "Biology 101", subject: subject } }
+  let!(:invalid_attributes) { { name: nil, subject: subject } }
```

Comment:
suggestion: These do not need to force invocation with a `!`, they can be simple `let` statements. This holds true in other specs as well.